docs/current_docs/api/interfaces.mdx

```diff
@@ -26,3 +36,77 @@ to include a `context.Context` argument.
 
 Note that you must also provide argument names, since they directly translate
 to the GraphQL field argument names.
+
+</TabItem>
+
+<TabItem value="TypeScript">
+
+The TypeScript SDK supports interfaces, which allow you to define set of functions
+that a object must implement to be considered a valid instance of that interface.
+That way, your module can accept arbitrary values from other modules without being
+tightly coupled to the concrete type underneath.
+
+To use an interface, use the TypeScript `interface` keyword and use it
+as a function argument:
+
+Here is an example of the definition of an interface `Fooer` with a single function `foo`:
+
+```ts file=./snippets/interfaces/declarations/typescript/index.ts
+```
+
+Functions defined in interface definitions must match the client-side API
+signature style:
+- Always define async function in interface (wrap the return type in a `Promise<T>`).
+- You can declare it as a method signature (e.g., `foo(): Promise<string>`) or a property signature (e.g., `foo: () => Promise<string>`).
```

Comment:
```suggestion
- Declare it as a method signature (e.g., `foo(): Promise<string>`) or a property signature (e.g., `foo: () => Promise<string>`).
```

`it` is ambiguous above, can you specify what `it` refers to?

Reply:
It's the function of the interface